### ISSUE TITLE
Invalid string to numeric type yields undefined behavior.

### ISSUE THREAD

**Issue by DylanCheetah** [godotengine/godot#99601]
### Tested versions

- Reproducible in: 4.3, 3.x

### System information

Windows 11 - Godot 4.3.stable - Forward+ - Intel Iris Xe

### Issue description

Attempting to cast a string to a numeric type yields undefined behavior instead of an error.

### Steps to reproduce

Run the project and the output will print numbers parsed from the invalid string to numeric type casts that don't reflect the string contents.

### Minimal reproduction project (MRP)

[NumericCastBug.zip](https://github.com/user-attachments/files/17882623/NumericCastBug.zip)

**Comment by Nallebeorn** [comment]
In the MRP, `int("t1ex2t")` prints `12` and `print(float("t1ex2t"))` prints `0.0`. This works as documented, so it's not really undefined behaviour (although arguably a bit unintuitive). The string constructors are documented as following the same rules as `String.to_int()` and `String.to_float()`, which in turn say:

> Converts the string representing an integer number into an [int](https://docs.godotengine.org/en/stable/classes/class_int.html#class-int). This method removes any non-number character and stops at the first decimal point (.).

> Converts the string representing a decimal number into a [float](https://docs.godotengine.org/en/stable/classes/class_float.html#class-float). This method stops on the first non-number character, except the first decimal point (.) and the exponent letter (e). See also [is_valid_float()](https://docs.godotengine.org/en/stable/classes/class_string.html#class-string-method-is-valid-float).

The docs even give an example that a string with no digits like "Hello" will return `0.0`, just like `float("t1ex2t")` does. The 3.6 docs are a bit less detailed, but still say something similar (including "Calling this method with an invalid float string will return 0").

If we want to change this behaviour or introduce some warning, then perhaps that would be more appropriate as a proposal than a bug report? @AThousandShips

**Comment by AThousandShips** [comment]
Indeed, please suggest any changes in a [proposal](https://github.com/godotengine/godot-proposals)
